src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.EnumeratedValueSet.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static partial class ValueSetFactory
+    {
+        /// <summary>
+        /// A value set that only supports equality and works by including or excluding specific values.
+        /// </summary>
+        private class EnumeratedValueSet<T, TTC> : IValueSet<T> where TTC : struct, EqualableValueTC<T>
```

Comment:
>EnumeratedValueSet [](start = 22, length = 18)

It would be worth mentioning that this is used for `string` and `decimal`, and explain why.

Reply:
Decimal will be a numeric one, it just isn't implemented until the next PR.  I'll add a comment explaining how and why this is used.

---
In reply to: [381028143](https://github.com/dotnet/roslyn/pull/41503#discussion_r381028143) [](ancestors = 381028143)